File: interfaceTest/ssros_ams/src/gesture_publisher.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "test_interfaces/msg/head_touch_gesture.hpp"
#include "builtin_interfaces/msg/time.hpp"

using namespace std::chrono_literals;

class GesturePublisher : public rclcpp::Node
{
public:
  GesturePublisher()
  : Node("gesture_publisher"), count_(0)
  {
    publisher_ = this->create_publisher<test_interfaces::msg::HeadTouchGesture>("head_touch_gesture", 10);
    timer_ = this->create_wall_timer(
      1s, std::bind(&GesturePublisher::timer_callback, this));
  }

private:
  void timer_callback()
  {
    auto message = test_interfaces::msg::HeadTouchGesture();
    message.stamp = this->get_clock()->now();
    message.gesture = count_ % 7;
    message.value = count_ % 2;
    message.direction = count_ % 3;
    
    RCLCPP_INFO(this->get_logger(), "Publishing: gesture=%d, value=%d, direction=%d", message.gesture, message.value, message.direction);
    publisher_->publish(message);
    count_++;
  }
  rclcpp::TimerBase::SharedPtr timer_;
  rclcpp::Publisher<test_interfaces::msg::HeadTouchGesture>::SharedPtr publisher_;
  size_t count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<GesturePublisher>());
  rclcpp::shutdown();
  return 0;
}

File: interfaceTest/ams/ssros_ams_py/ssros_ams_py/gesture_publisher.py
import rclpy
from rclpy.node import Node
from test_interfaces.msg import HeadTouchGesture
from builtin_interfaces.msg import Time


class GesturePublisher(Node):
    def __init__(self):
        super().__init__("gesture_publisher_py")
        self.publisher_ = self.create_publisher(
            HeadTouchGesture, "head_touch_gesture", 10
        )
        self.timer = self.create_timer(1.0, self.timer_callback)
        self.count_ = 0

    def timer_callback(self):
        msg = HeadTouchGesture()
        msg.stamp = self.get_clock().now().to_msg()
        msg.gesture = self.count_ % 7
        msg.value = self.count_ % 2
        msg.direction = self.count_ % 3

        self.get_logger().info(
            "Publishing: gesture=%d, value=%d, direction=%d"
            % (msg.gesture, msg.value, msg.direction)
        )
        self.publisher_.publish(msg)
        self.count_ += 1


def main(args=None):
    rclpy.init(args=args)
    gesture_publisher = GesturePublisher()
    rclpy.spin(gesture_publisher)
    gesture_publisher.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()
